spring-cloud-contract-wiremock/src/main/java/org/springframework/cloud/contract/wiremock/WireMockRestTemplateConfiguration.java

```diff
@@ -48,11 +56,20 @@ public RestTemplateCustomizer wiremockRestTemplateCustomizer() {
 		return new RestTemplateCustomizer() {
 			@Override
 			public void customize(RestTemplate restTemplate) {
-				if (restTemplate.getRequestFactory() instanceof HttpComponentsClientHttpRequestFactory) {
-					HttpComponentsClientHttpRequestFactory factory = (HttpComponentsClientHttpRequestFactory) restTemplate
-							.getRequestFactory();
+				if (restTemplate.getRequestFactory() instanceof HttpComponentsClientHttpRequestFactory factory) {
 					factory.setHttpClient(createSslHttpClient());
 				}
+				else if (restTemplate.getRequestFactory() instanceof InterceptingClientHttpRequestFactory) {
```

Comment:
Is there any other way rather than doing this via reflection? Maybe we should ask the Spring team for relaxing some constraints there if there's no other way?

Reply:
I was not able to find better way. I will create ticket in Spring framework